### PR TITLE
Lint: Upgrade Lint Baseline

### PR DESCRIPTION
**Pull request by ParaskP7**
This PR upgrades the lint baseline to latest `5` format, `4.0.2` version. This is done in order to make lint more stable and up-to-date.

To test:
1. Pull the branch locally.
2. Run ` ./gradlew lintWordPressVanillaRelease` and verify everything works as expected.
3. Check CI and verify everything works as expected.

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I run `Lint` in both, `debug` and `release` build types, as well as both, `wordPress` and `jetpack` dimensions to make sure that this change is not affecting the lint report and the actual reported issues, both `errors` and `warnings`.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ParaskP7** [comment]
👋 @oguzkocer and @AliSoftware !

Everything seems to be working ok, but once again I am stumbling at the below `Local` issue that I can't seem to figure out. I wonder why how comes `CI` succeeds and locally I am failing. I have been having this issue for as long as I remember, I was thinking that the baseline will solve it, I also used a different `WordPress-Android` clone to start fresh, but it seems that I am still having discrepancies. PS: I am sure others have too.

Is there anything I am missing here? PS: I'll continue to investigate further in the meanwhile.

So, in summary, locally I have:
- 156 additional errors, 2 `MissingPermission` and 154 `Range` errors.
- 10 extra warnings, on the `GradleDependency` warning.

### CI vs Local 

Where | Console Output
-----|-----
CI | lintWordpressVanillaRelease (Console - CI)
Local | lintWordpressVanillaRelease (Console - Local)  

CI Report | Local Report
-----|-----
ci | local

**Comment by AliSoftware** [comment]
I ran the lint on my Mac and got the same as what you screenshoted for the CI case:

```
$ ./gradlew lintWordPressVanillaRelease
…

Lint found 335 warnings (23 errors filtered by baseline lint-baseline.xml)
```

I'm also puzzled as to what could lead to different results here. Oguz mentioned that the diff between local and CI might be a different version of _something_ (JVM? Gradle? Detekt?) but not sure what…

**Comment by AliSoftware** [comment]
Mmmmh wait I have the same total… but a different summary per type apparently – for example I have 48 `GradleDependency` warnings while your screenshots show 43 (CI) / 53 (Local) 🤔 

LintReport

**Comment by ParaskP7** [comment]
👋 @AliSoftware !

Thanks so much for going through the trouble in running it locally and posting the results. 🙏 

I believe that this non-deterministic nature of the results is problematic and that we should try to investigate it further to figure out why that is the case. 🤔

I'll keep exploring and will be posting my finding here... 🧾 

PS: This `GradleDependency` discrepancy might be giving us a hint of where the problem lies. 🤷

**Comment by oguzkocer** [comment]
@ParaskP7 I've a suggestion. How about we try and fix the issues in `lint-baseline.xml` and remove the file completely? I am not sure if it's worth spending any more time on this file. I feel that time would be better spent by fixing the issues instead.

The discrepancy between different local builds and CI is definitely a problem we should address, but I'd rather do that after all the build file cleanups and Gradle upgrades are done. The problem might even go away as part of that work as I intend to fix the Gradle related lint issues anyway.

Let me know how you'd like to proceed!

**Comment by ParaskP7** [comment]
👋 @oguzkocer !

> @ParaskP7 I've a suggestion. How about we try and fix the issues in lint-baseline.xml and remove the file completely? I am not sure if it's worth spending any more time on this file. I feel that time would be better spent by fixing the issues instead.

I agree. I can close this PR and open another one where I will be fixing those instead of converting this baseline to the new format. However, the conversion wasn't that hard now that the issues became a handful, it took me just a bit of time to do.

> The discrepancy between different local builds and CI is definitely a problem we should address, but I'd rather do that after all the build file cleanups and Gradle upgrades are done. The problem might even go away as part of that work as I intend to fix the Gradle related lint issues anyway.

This is true, let's do that next, it sounds more reasonable than me hunting down this discrepancy with the current state.

FYI: What I figured out so far is that those errors that I am getting is due to newer `SDK` versions, currently we are on 29 (Android 10), but by moving to 30 (Android 11) or even 31 (Android 12), then those errors will end up popping-up to everyone. I am still however not sure why I am the only one getting them. Maybe it is because I also have Beta and Canary versions of Android Studio, which bring new tooling with them. However, I wouldn't expect that this will conflict with the usage of the Stable version, and Gradle CLI for that matter, which should be IDE agnostic. 🤷 

PS To try that out, just Go to line `88` of the WordPress module's `build.gradle` file and change it to `compileSdkVersion "android-S"`. Then, navigate to, for example `ShareAndDismissNotificationReceiver.java` class, and notice this Lint error now showing there as well.

> Let me know how you'd like to proceed!

Let's do as you say, close this PR and continue by removing the individual issues from the baseline, with the end goal removing the baseline for good.

**Comment by ParaskP7** [comment]
> PS To try that out, just Go to line 88 of the WordPress module's build.gradle file and change it to compileSdkVersion "android-S". Then, navigate to, for example ShareAndDismissNotificationReceiver.java class, and notice this Lint error now showing there as well.

**UPDATE**: I actually managed to successfully build Lint, without those extra errors I was getting, by:
- First removing everything `31` related from the `SDK Tools` in AS (`Android SDK`),
- Then, remove the `Android S Preview` from the `SDK Platforms` in AS (`Android SDK`).

That did the trick in the end! 🥳

**Comment by ParaskP7** [comment]
👋 @oguzkocer and @AliSoftware !

As discussed, I am closing the PR now and we will deal with this problem by fixing the remaining issues instead and removing this `Lint` baseline file for good.
